packages/SwingSet/src/controller.js

```diff
@@ -204,11 +204,19 @@ function realmRegisterEndOfCrank(fn) {
     // static vats once we add metering support to the dynamic vat
     // implementation.
     // FIXME: Same for registerEndOfCrank.
-    return s.evaluate(src, {
-      require: r,
-      registerEndOfCrank: realmRegisterEndOfCrank,
-      replaceGlobalMeter,
-    })().default;
+
+    // Cope with ESM problems.
+    src = src.replace(/(_[a-zA-Z0-9]{3}\u200d\.e)/g, 'eval');
+
+    // Support both getExport and nestedEvaluate module format.
+    const nestedEvaluate = source =>
+      s.evaluate(source, {
+        require: r,
+        registerEndOfCrank: realmRegisterEndOfCrank,
+        replaceGlobalMeter,
+        nestedEvaluate,
+      });
+    return nestedEvaluate(src)().default;
```

Comment:
Inject the vatID or something instead of `()` for the `"/bundled-source"` source string replacement.